tests/unittests/Types_test.cpp

```diff
@@ -130,6 +138,16 @@ TEST(AVariant, ReturnsTrueWhenAskedIfItContainsTheTypeItReallyContains)
     ASSERT_TRUE(variant.containsValueOfType<ComplexType>());
 }
 
+TEST(AVariant, CanBeConvertedIntoAnStdVariant)
+{
+    using VariantType = std::variant<std::string, uint64_t, std::vector<sdbus::Struct<std::string, double>>>;
+    VariantType value{ std::vector<sdbus::Struct<std::string, double>>{{"hello"s, ANY_DOUBLE}, {"world"s, ANY_DOUBLE}} };
+
+    sdbus::Variant variant(value);
+
+    ASSERT_EQ(value, VariantType { variant });
```

Comment:
Would you please change to:

```suggestion
    ASSERT_THAT(value, Eq(VariantType{variant}));
```

We use this Hamcrest notation in sdbus-c++ tests as we find it more readable and clear what is being tested against what (the only exception is testing simply boolean expressions, using `ASSERT_TRUE()`/`ASSERT_FALSE()`)

Also minor, but I would rename `VariantType` to `StdVariantType` for better clarity...